### PR TITLE
compatible with openai/tgi/vllm request format

### PR DESCRIPTION
**Pull request by lkk12014402**
## Description

1. compatible with openai/tgi/vllm `completion` request format/parameters
2. priority default values (openai > tgi > vllm). With same default value, the opea service latency is almost same with native server tgi/vllm
3. replace `from langchain_community.llms import VLLMOpenAI` with `from openai import OpenAI`, which can reduce initialization time (0.15s -> 0.01s) and only init once
4. simplify post-process for openai format response (especially remove `encode`)

### CODE REVIEW

**Comment by lkk12014402** [comment]
| **OpenAI **[API Reference - OpenAI API](https://platform.openai.com/docs/api-reference/chat/create) | **vLLM** [vllm/openai/protocol.py](https://github.com/vllm-project/vllm/blob/main/vllm/entrypoints/openai/protocol.py#L129) | **TGI** [GenerateParameters](https://github.com/huggingface/text-generation-inference/blob/main/router/src/lib.rs#L190) | **Opea (old)** | **Opea (this pr)** |
| ------------------------------------------------------------ | ------------------------------------------------------------ | ------------------------------------------------------------ | -------------- | :----------------: |
| model: Union[str]                                            |                                                              |                                                              |                |                    |
|                                                              |                                                              |                                                              |                |                    |
|                                                              |                                                              |                                                              |                |                    |
|                                                              |                                                              |                                                              |                |                    |

**Comment by lkk12014402** [comment]
### completion request format

<html xmlns:o="urn:schemas-microsoft-com:office:office"
xmlns:dt="uuid:C2F41010-65B3-11d1-A29F-00AA00C14882"
xmlns="http://www.w3.org/TR/REC-html40">

<head>

<meta name=ProgId content=OneNote.File>
<meta name=Generator content="Microsoft OneNote 15">
</head>

<body lang=en-US style='font-family:Calibri;font-size:11.0pt'>


<div style='direction:ltr'>


OpenAI API Reference - OpenAI API | vLLM vllm/entrypoints/openai/protocol.py | TGI GenerateParameters | OPEA (now) | OPEA (this pr)
-- | -- | -- | -- | --
model:   Union[str | model:   str |   |   | model:   Optional[str] = None # for openai, not used by tgi
prompt:   Union[str | prompt:   Union[List[int], List[List[int]], str, List[str]] |   |   | query:   str # alias 'prompt'
best_of:   Optional[int]   Defaults to 1 | best_of:   Optional[int] = None | default   = "null" |   | best_of:   Optional[int] = 1
echo:   Optional[bool]   Defaults to false | echo:   Optional[bool] = False |   |   | echo:   Optional[bool] = False
frequency_penalty: Optional[float]   Defaults to 0 | frequency_penalty:   Optional[float] = 0.0 | default   = "null" | frequency_penalty:   Optional[float] = 0.0 | frequency_penalty:   Optional[float] = 0.0
logit_bias:   Optional[Dict[str, int]]   Defaults to null | logit_bias:   Optional[Dict[str, float]] = None |   |   | logit_bias:   Optional[Dict[str, float]] = None
logprobs:   Optional[int]   Defaults to null | logprobs:   Optional[int] = None |   |   | logprobs:   Optional[int] = None
max_tokens: Optional[int]   Defaults to 16 | max_tokens:   Optional[int] = 16 | default   = "100" | max_tokens:   Optional[int] = 1024 | max_new_tokens:   Optional[int] = 16  # alias   'max_tokens'
n: Optional[int]   Defaults to 1 | n: int   = 1 |   |   | n:   Optional[int] = 1
presence_penalty: Optional[float]   Defaults to 0 | presence_penalty:   Optional[float] = 0.0 |   |   | presence_penalty:   Optional[float] = 0.0
seed:   Optional[int]   Defaults to null | seed: Optional[int] = Field(None, | default   = "null" |   | seed:   Optional[int] = None
stop:   Union[Optional[str]   Defaults to null | stop:   Optional[Union[str, List[str]]] = Field(default_factory=list) | stop:   Vec<String>, default = [] |   | stop:   Union[Optional[str], List[str], None] = None
stream:   Optional[Literal[False]]   Defaults to false | stream:   Optional[bool] = False |   | stream:   Optional[bool] = False | streaming:   Optional[bool] = False  # alias   'stream'
stream_options:   Optional[   Defaults to null | stream_options:   Optional[StreamOptions] = None |   |   | stream_options:   Optional[StreamOptions] = None
suffix:   Optional[str]   Defaults to null | suffix:   Optional[str] = None |   |   | suffix:   Optional[str] = None
temperature:   Optional[float]   Defaults to 1 | temperature:   Optional[float] = 1.0 | default   = "null", | temperature:   Optional[float] = 0.01 | temperature:   Optional[float] = 1.0
top_p:   Optional[float]   Defaults to 1 | top_p:   Optional[float] = 1.0 | default   = "null", | top_p:   Optional[float] = 0.95 | top_p:   Optional[float] = 1.0
user:   str   Defaults to null | user:   Optional[str] = None |   |   | user:   Optional[str] = None
  | more parameters for vllm….. |   |   |  
  | … | default   = "null" | top_k:   Optional[int] = 10 | top_k:   Optional[int] = None
  | … | default   = "null" | typical_p:   float = 0.95 | typical_p:   Optional[float] = None
  | … | default   = "null" | repetition_penalty:   Optional[float] = 1.03 | repetition_penalty:   Optional[float] = None
  |   | do_sample:   bool, default =   "false" |   |  



</div>


</body>

</html>
OpenAI [API Reference - OpenAI API](https://platform.openai.com/docs/api-reference/chat/create)	vLLM [vllm/entrypoints/openai/protocol.py](https://github.com/vllm-project/vllm/blob/main/vllm/entrypoints/openai/protocol.py#L359)	TGI [GenerateParameters](https://github.com/huggingface/text-generation-inference/blob/main/router/src/lib.rs#L190)	OPEA (now)	OPEA (this pr)
model: Union[str	model: str			model: Optional[str] = None # for openai, not used by tgi
prompt: Union[str	prompt: Union[List[int], List[List[int]], str, List[str]]			query: str # alias 'prompt'
best_of: Optional[int]	best_of: Optional[int] = None	default = "null"		best_of: Optional[int] = 1
Defaults to 1
echo: Optional[bool]	echo: Optional[bool] = False			echo: Optional[bool] = False
Defaults to false
 frequency_penalty: Optional[float]	frequency_penalty: Optional[float] = 0.0	default = "null"	frequency_penalty: Optional[float] = 0.0	frequency_penalty: Optional[float] = 0.0
Defaults to 0
logit_bias: Optional[Dict[str, int]]	logit_bias: Optional[Dict[str, float]] = None			logit_bias: Optional[Dict[str, float]] = None
Defaults to null
logprobs: Optional[int]	logprobs: Optional[int] = None			logprobs: Optional[int] = None
Defaults to null
 max_tokens: Optional[int]	max_tokens: Optional[int] = 16	default = "100"	max_tokens: Optional[int] = 1024	max_new_tokens: Optional[int] = 16  # alias 'max_tokens'
Defaults to 16
 n: Optional[int]	n: int = 1		 	n: Optional[int] = 1
Defaults to 1
 presence_penalty: Optional[float]	presence_penalty: Optional[float] = 0.0			presence_penalty: Optional[float] = 0.0
Defaults to 0
seed: Optional[int]	seed: Optional[int] = Field(None,	default = "null"		seed: Optional[int] = None
Defaults to null
stop: Union[Optional[str]	stop: Optional[Union[str, List[str]]] = Field(default_factory=list)	stop: Vec<String>, default = []		stop: Union[Optional[str], List[str], None] = None
Defaults to null
stream: Optional[Literal[False]]	stream: Optional[bool] = False		stream: Optional[bool] = False	streaming: Optional[bool] = False  # alias 'stream'
Defaults to false
stream_options: Optional[	stream_options: Optional[StreamOptions] = None			stream_options: Optional[StreamOptions] = None
Defaults to null
suffix: Optional[str]	suffix: Optional[str] = None			suffix: Optional[str] = None
Defaults to null
temperature: Optional[float]	temperature: Optional[float] = 1.0	default = "null",	temperature: Optional[float] = 0.01	temperature: Optional[float] = 1.0
Defaults to 1
top_p: Optional[float]	top_p: Optional[float] = 1.0	default = "null",	top_p: Optional[float] = 0.95	top_p: Optional[float] = 1.0
Defaults to 1
user: str	user: Optional[str] = None			user: Optional[str] = None
Defaults to null
	more parameters for vllm…..			
	…	default = "null"	top_k: Optional[int] = 10	top_k: Optional[int] = None
	…	default = "null"	typical_p: float = 0.95	typical_p: Optional[float] = None
	…	default = "null"	repetition_penalty: Optional[float] = 1.03	repetition_penalty: Optional[float] = None
		do_sample: bool, default = "false"

File: comps/llms/text-generation/vllm-xft/llm.py [hunk 1]
```diff
@@ -19,33 +28,43 @@
 )
 @traceable(run_type="llm")
 def llm_generate(input: LLMParamsDoc):
-    llm_endpoint = os.getenv("vLLM_LLM_ENDPOINT", "http://localhost:18688")
-    llm = VLLMOpenAI(
-        openai_api_key="EMPTY",
-        openai_api_base=llm_endpoint + "/v1",
+
```
Comment (ftian1) [inline comps/llms/text-generation/vllm-xft/llm.py:31]: pls also update other vllm instances under llms/text_generation/

**Review by lvliang-intel — APPROVED** [review]

File: comps/llms/text-generation/vllm-xft/llm.py (hunk 1 above)
Comment (lkk12014402) [reply comps/llms/text-generation/vllm-xft/llm.py:31]: yes, asap

**Comment by lkk12014402** [comment]
implement it in another pr

**Pull request closed without merging**